CMakeLists.txt

```diff
@@ -16,7 +16,7 @@ endif()
 #-----------------------------------------------------------------------
 # - Define CMake requirements and override make rules as needed
 #
-cmake_minimum_required(VERSION 3.8 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
```

Comment:
Is there a way to allow the functionality with CMake 3.8? We generally don't change CMake version in older versions, so a "back port" would be better.

Alternately, can the change in G4BuildSettings be protected with a check on CMAKE_VERSION, erroring out if the requirement of 3.15 or better isn't met?

Reply:
Before CMake 3.15, the approach is to manually add the relevant flags (/MT or /MTd) to each of the CMAKE_CXX_FLAGS_XXX variables. Alternatively, users with CMake >= 3.15 can manually override the policy (as described in the link I posted above).

In summary, my pull request is an improvement to consider when (if) CMake is bumped to 3.15.

Reply:
Thanks (and sorry for the delay in replying)!  I'll double check that we're doing the right thing with the flags, and see if we can use the newer method when the client cmake is 3.15 (whilst still having 3.8 as the minimum).

Reply:
I think that it is fine as it is  now :-)